src/components/Stepper/Stepper.tsx

```diff
@@ -0,0 +1,223 @@
+    stepsArr:string[],
+    detectError?(step: number): boolean,
+    detectOptional?(step: number) : boolean,
+    customErrorSvg?: React.ReactNode,
+    customFinishedSvg?: React.ReactNode,
+    children?: React.ReactNode[],
+}
+
+
+export const Stepper: FC<IStepperProps> = ({
+                                               horizontalContainerClassName,
+                                               verticalContainerClassName,
+                                               horizontalItemClassName,
+                                               horizontalContentClassName,
+                                               verticalItemStyle,
+                                               iconContainerStyle,
+                                               verticalTitleStyle,
+                                               horizontalLabelClassName,
+                                               horizontalTitleStyle,
+                                               buttonGroupClassName,
+                                               backButtonClassName,
+                                               resetButtonClassName,
+                                               skipButtonClassName,
+                                               continueButtonClassName,
+                                               alternative=false,
+                                               verticalStepper=false,
+                                               stepsArr,
+                                               detectError,
+                                               detectOptional,
+                                               customErrorSvg,
+                                               customFinishedSvg,
+                                               children}) => {
+
+    const [activeStep, setActiveStep] = useState<number>(0);
+    const [skipped, setSkipped] = useState(new Set<number>());
+    const [errorSteps, setErrorSteps] = useState(new Set<number>());
+
+
+    const checkIsOptional = (step: number): boolean => {
+        if (detectOptional) {
+            return detectOptional(step);
+        } else {
+            return false;
+        }
+    }
+
+    const checkIsError = (step: number) => {
+        if (detectError) {
+            return detectError(step);
+        } else {
+            return false;
+        }
+    }
+
+    const checkIsSKipped = (step: number) => {
+        return skipped.has(step);
+    }
+
+    const handleSkipClick = () => {
+        if (!checkIsOptional(activeStep)) {
+            throw new Error('You can\'t skip a step that is not optional.')
+        }
+        setActiveStep(prevState => prevState + 1);
+        setSkipped(prevState => {
+            const newSkipped = new Set(prevState.values())
+            newSkipped.add(activeStep);
+            return newSkipped;
+        })
+    }
+
+    const handleBackClick = () => {
+        setActiveStep(prevState => prevState - 1);
+    }
```

Comment:
handleBackClick should remove previous step from skipped and errorSteps. Otherwise, the button would still have the error state and the skipped state if the user goes back to the previous step after skipping or making an error.